### PR TITLE
Add missing supportCompression parameter to NIOServerConfig object

### LINKED ISSUE #1907: Add missing supportCompression parameter to NIOServerConfig object
Add missing `supportCompression` parameter to NIOServerConfig object to enable compression in `HTTPServer.start` method. `HTTPServer.start` method with `supportCompression` parameter implemented here https://github.com/vapor/http/issues/280

### PR DESCRIPTION
**Pull request by marekpridal**
Closes #1907

Added missing supportCompression parameter to NIOServerConfig. Can be set in `init`, `default` or directly on object (is `var`).

### Checklist

- [ ] Circle CI is passing (code compiles and passes tests).
- [x] There are no breaking changes to public API.
- [ ] New test cases have been added where appropriate.
- [x] All new code has been commented with doc blocks `///`.

### CODE REVIEW

File: Sources/Vapor/Server/NIOServerConfig.swift [hunk 1]
```diff
@@ -30,6 +30,7 @@ public struct NIOServerConfig: ServiceType {
         maxBodySize: Int = 1_000_000,
         reuseAddress: Bool = true,
         tcpNoDelay: Bool = true,
+        supportCompression: Bool = false,
```
Comment (tanner0101) [inline Sources/Vapor/Server/NIOServerConfig.swift]: New parameter should be added to the docblock.

Comment (marekpridal) [reply Sources/Vapor/Server/NIOServerConfig.swift]: Good point 👍🏻I’ll do it tomorrow, it’s nearly midnight in Prague 😅 BTW what’s the best way how to send fix up of PR? It’s only my third PR and first time I need to send fix up 🙂

Comment (marekpridal) [reply Sources/Vapor/Server/NIOServerConfig.swift]: Find out it should be OK to push fix to same branch and GitHub should add it to this PR. I’ll try it tomorrow 🙂

Comment (marekpridal) [reply Sources/Vapor/Server/NIOServerConfig.swift]: Fixed in `31cd94e`

**Comment by penny-coin** [comment]
Hey @marekpridal, you just merged a pull request, have a coin! 

You now have 1 coins.

**Pull request merged**